pkg/suggestion/v1beta1/hyperopt/base_service.py

```diff
@@ -63,13 +64,66 @@ def create_hyperopt_domain(self):
         hyperopt_search_space = {}
         for param in self.search_space.params:
             if param.type == INTEGER:
-                hyperopt_search_space[param.name] = hyperopt.hp.quniform(
-                    param.name, float(param.min), float(param.max), float(param.step)
-                )
-            elif param.type == DOUBLE:
-                hyperopt_search_space[param.name] = hyperopt.hp.uniform(
+                hyperopt_search_space[param.name] = hyperopt.hp.uniformint(
                     param.name, float(param.min), float(param.max)
                 )
+            elif param.type == DOUBLE:
+                if param.distribution == api_pb2.UNIFORM or param.distribution is None:
+                    if param.step:
+                        hyperopt_search_space[param.name] = hyperopt.hp.quniform(
+                            param.name,
+                            float(param.min),
+                            float(param.max),
+                            float(param.step),
+                        )
+                    else:
+                        hyperopt_search_space[param.name] = hyperopt.hp.uniform(
+                            param.name, float(param.min), float(param.max)
+                        )
+                elif param.distribution == api_pb2.LOG_UNIFORM:
+                    if param.step:
+                        hyperopt_search_space[param.name] = hyperopt.hp.qloguniform(
+                            param.name,
+                            float(param.min),
+                            float(param.max),
+                            float(param.step),
+                        )
+                    else:
+                        hyperopt_search_space[param.name] = hyperopt.hp.loguniform(
+                            param.name, float(param.min), float(param.max)
+                        )
+                elif param.distribution == api_pb2.NORMAL:
+                    mu = (float(param.min) + float(param.max)) / 2
+                    sigma = (float(param.max) - float(param.min)) / 6
```

Comment:
I followed this [article](https://soc.utah.edu/sociology3112/normal-distribution.php) to determine the value of `sigma` from `min` and `max`.

Reply:
Maybe we should add this article to the comments. WDYT @tenzen-y @johnugeorge ?

Reply:
I do not want to depend on the individual article. Instead of that, it would be better to add an actual mathematical description here as a comment.